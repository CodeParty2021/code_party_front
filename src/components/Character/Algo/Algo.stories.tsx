import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Algo } from "./Algo";

export default {
  title: "components/Algo",
  component: Algo,
} as ComponentMeta<typeof Algo>;

const Template: ComponentStory<typeof Algo> = (args) => <Algo {...args} />;

export const Default = Template.bind({});
Default.args = {
  color: "turquoise",
};
