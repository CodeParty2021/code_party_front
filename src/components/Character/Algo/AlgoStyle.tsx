import { ReactSVG } from "react-svg";
import styled, { css } from "styled-components";
import { algo } from "styles/colors";

export type AlgoStyleProps = {
  color?: "turquoise" | "leaf" | "orange" | "magenta";
};

const defaultStyle = css<AlgoStyleProps>``;

const turquoiseStyle = css`
  .algo_color_main {
    fill: ${algo.turquoise};
  }
`;

const leafStyle = css`
  .algo_color_main {
    fill: ${algo.leaf};
  }
`;

const orangeStyle = css`
  .algo_color_main {
    fill: ${algo.orange};
  }
`;

const magentaStyle = css`
  .algo_color_main {
    fill: ${algo.magenta};
  }
`;

export const AlgoStyle = styled(ReactSVG)<AlgoStyleProps>`
  ${defaultStyle}

  ${({ color }) => color == "turquoise" && turquoiseStyle}
  ${({ color }) => color == "leaf" && leafStyle}
  ${({ color }) => color == "orange" && orangeStyle}
  ${({ color }) => color == "magenta" && magentaStyle}
`;

AlgoStyle.defaultProps = {
  color: "turquoise",
};
