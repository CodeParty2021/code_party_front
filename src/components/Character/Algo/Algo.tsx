import React from "react";
import { AlgoStyle, AlgoStyleProps } from "./AlgoStyle";

type Props = AlgoStyleProps & {};

export const Algo: React.FC<Props> = ({ ...styleProps }) => {
  return <AlgoStyle src="/img/algo.svg" {...styleProps} />;
};
