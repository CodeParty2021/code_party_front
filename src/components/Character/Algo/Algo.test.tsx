import React from "react";
import { Algo } from "./Algo";
import "jest-styled-components";
import { render } from "enzyme";

describe("<Algo />", () => {
  it("snapshot test", () => {
    const wrapper = render(<Algo />);

    expect(wrapper).toMatchSnapshot();
  });
});
