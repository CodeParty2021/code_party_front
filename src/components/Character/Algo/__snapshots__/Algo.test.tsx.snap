// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Algo /> snapshot test 1`] = `
.c0 .algo_color_main {
  fill: #34C5D5;
}

<div
  class="c0"
  color="turquoise"
/>
`;
